network/netvisor/pn_ospf.py

```diff
@@ -0,0 +1,283 @@
+        cli += ' switch-local '
+    else:
+        cli += ' switch ' + cliswitch
+    return cli
+
+
+def check_cli(module, cli):
+    """
+    This method checks if vRouter exists on the target node.
+    This method also checks for idempotency using the vrouter-ospf-show command.
+    If the given vRouter exists, return VROUTER_EXISTS as True else False.
+    If an OSPF network with the given ip exists on the given vRouter,
+    return NETWORK_EXISTS as True else False.
+
+    :param module: The Ansible module to fetch input parameters
+    :param cli: The CLI string
+    :return Global Booleans: VROUTER_EXISTS, NETWORK_EXISTS
+    """
+    vrouter_name = module.params['pn_vrouter_name']
+    network_ip = module.params['pn_network_ip']
+    # Global flags
+    global VROUTER_EXISTS, NETWORK_EXISTS
+
+    # Check for vRouter
+    check_vrouter = cli + ' vrouter-show format name no-show-headers '
+    check_vrouter = shlex.split(check_vrouter)
+    out = module.run_command(check_vrouter)[1]
+    out = out.split()
+
+    if vrouter_name in out:
+        VROUTER_EXISTS = True
+    else:
+        VROUTER_EXISTS = False
+
+    # Check for OSPF networks
+    show = cli + ' vrouter-ospf-show vrouter-name %s ' % vrouter_name
+    show += 'format network no-show-headers'
+    show = shlex.split(show)
+    out = module.run_command(show)[1]
+    out = out.split()
+
+    if network_ip in out:
+        NETWORK_EXISTS = True
+    else:
+        NETWORK_EXISTS = False
+
+
+def run_cli(module, cli):
+    """
+    This method executes the cli command on the target node(s) and returns the
+    output. The module then exits based on the output.
+    :param cli: the complete cli string to be executed on the target node(s).
+    :param module: The Ansible module to fetch command
+    """
+    cliswitch = module.params['pn_cliswitch']
+    command = module.params['pn_command']
+    cmd = shlex.split(cli)
+    response = subprocess.Popen(cmd, stderr=subprocess.PIPE,
+                                stdout=subprocess.PIPE, universal_newlines=True)
+    # 'out' contains the output
+    # 'err' contains the error messages
+    out, err = response.communicate()
+
+    print_cli = cli.split(cliswitch)[1]
+
+    # Response in JSON format
+    if err:
+        module.exit_json(
+            command=print_cli,
+            stderr=err.strip(),
+            msg="%s operation failed" % command,
+            changed=False
+        )
+
+    if out:
+        module.exit_json(
+            command=print_cli,
+            stdout=out.strip(),
+            msg="%s operation completed" % command,
+            changed=True
+        )
+
+    else:
+        module.exit_json(
+            command=print_cli,
+            msg="%s operation completed" % command,
+            changed=True
+        )
+
+
+def main():
+    """ This section is for arguments parsing """
+    module = AnsibleModule(
+        argument_spec=dict(
+            pn_cliusername=dict(required=False, type='str'),
```

Comment:
Rather than repeating the same arguments (and documentation for those arguments) you should move this to common code in the github/com/ansible/ansible/ repo:

ansible/ansible/lib/ansible/utils/module_docs_fragments/ # Document here
ansible/ansible/lib/ansible/module_utils/ # Define here

Have a look at `ansible-modules-extra/networking/f5` for an example of how to do this


In the `DOCUMENATION:` section in this (and other) module you then do
`extends_documentation_fragment: `